### PR TITLE
bugfix_token-balance-cache

### PR DESCRIPTION
**Pull request by BaojunCZ**

### CODE REVIEW

File: app/src/main/java/org/nervos/neuron/activity/WalletManageActivity.kt [hunk 1]
```diff
@@ -79,7 +78,7 @@ class WalletManageActivity : NBaseActivity() {
 
     private fun updateWalletName(simpleDialog: SimpleDialog) {
         when {
-            TextUtils.isEmpty(simpleDialog.message) -> Toast.makeText(mActivity, R.string.wallet_name_not_null, Toast.LENGTH_SHORT).show()
+            TextUtils.isEmpty(simpleDialog.message) || simpleDialog.message.trim().isEmpty() -> Toast.makeText(mActivity, R.string.wallet_name_not_null, Toast.LENGTH_SHORT).show()
```
Comment (duanyytop) [inline app/src/main/java/org/nervos/neuron/activity/WalletManageActivity.kt]: Why two isEmpty ?

File: app/src/main/java/org/nervos/neuron/view/WalletTokenView.kt [hunk 2]
```diff
@@ -43,13 +44,20 @@ class WalletTokenView(context: Context, attrs: AttributeSet) : LinearLayout(cont
         this.address = address
         TokenLogoUtil.setLogo(tokenItem, context, iv_token_logo)
         tv_token_name.text = tokenItem.symbol
-        tv_loading.text = resources.getString(R.string.wallet_token_loading)
+        if (tokenItem.balance == -1.0) {
```
Comment (duanyytop) [inline app/src/main/java/org/nervos/neuron/view/WalletTokenView.kt]: Why -1.0?

File: app/src/main/java/org/nervos/neuron/util/db/DBWalletUtil.java [hunk 3]
```diff
@@ -242,6 +246,21 @@ public static void addTokenToWallet(Context context, String walletName, TokenIte
...
+    public static void saveTokenBalanceCacheToWallet(Context context, String walletName, TokenItem tokenItem) {
+        WalletItem walletItem = getWallet(context, walletName);
+        if (walletItem != null) {
+            if (walletItem.tokenItems == null) {
+                walletItem.tokenItems = new ArrayList<>();
+            }
+            int index = checkTokenInWallet(walletItem, tokenItem);
+            if (index != -1) {
+                walletItem.tokenItems.remove(index);
+                walletItem.tokenItems.add(index, tokenItem);
```
Comment (duanyytop) [inline app/src/main/java/org/nervos/neuron/util/db/DBWalletUtil.java]: Update data can use List.set(index, newData)

File: app/src/main/java/org/nervos/neuron/util/db/DBWalletUtil.java [hunk 4]
```diff
@@ -42,9 +42,12 @@ public static WalletItem getCurrentWallet(Context context) {
     }
 
     public static WalletItem initChainToCurrentWallet(Context context, WalletItem walletItem) {
-        walletItem.chainItems.add(new ChainItem(ConstantUtil.ETHEREUM_MAIN_ID, ConstantUtil.ETH_MAINNET, ConstantUtil.ETH, ConstantUtil.ETH));
-        walletItem.chainItems.add(new ChainItem(ConstantUtil.CMB_CHAIN_ID, ConstantUtil.CMB_CHAIN_NAME, ConstantUtil.CMB_HTTP_PROVIDER, ConstantUtil.CMB_TOKEN_NAME, ConstantUtil.CMB_TOKEN_SYMBOL, ConstantUtil.CMB_TOKEN_AVATAR));
-        walletItem.chainItems.add(new ChainItem(ConstantUtil.NATT_CHAIN_ID, ConstantUtil.NATT_CHAIN_NAME, ConstantUtil.NATT_HTTP_PROVIDER, ConstantUtil.NATT_TOKEN_NAME, ConstantUtil.NATT_TOKEN_SYMBOL, ConstantUtil.NATT_TOKEN_AVATAR));
+        walletItem.chainItems.add(new ChainItem(ConstantUtil.ETHEREUM_MAIN_ID, ConstantUtil.ETH_MAINNET, ConstantUtil.ETH, ConstantUtil
+                .ETH));
```
Comment (duanyytop) [inline app/src/main/java/org/nervos/neuron/util/db/DBWalletUtil.java]: ConstantUtil.ETH should be written to one line

File: app/src/main/java/org/nervos/neuron/activity/WalletManageActivity.kt (hunk 1 above)
Comment (BaojunCZ) [reply app/src/main/java/org/nervos/neuron/activity/WalletManageActivity.kt]: The second check name what is all space character

File: app/src/main/java/org/nervos/neuron/view/WalletTokenView.kt (hunk 2 above)
Comment (BaojunCZ) [reply app/src/main/java/org/nervos/neuron/view/WalletTokenView.kt]: Tag of no cache

**Pull request merged**